src/main/java/org/dice_research/opal/civet/metrics/DataFormatMetric.java

```diff
@@ -0,0 +1,124 @@
+ * 
+ * @see https://www.w3.org/TR/vocab-dcat-1/#Property:distribution_format
+ * @see https://www.w3.org/TR/vocab-dcat-1/#Property:distribution_media_type
+ * 
+ * @author Gourab Sahu
+ */
+public class DataFormatMetric implements Metric {
+
+	private static final Logger LOGGER = LogManager.getLogger();
+	private static final String DESCRIPTION = 
+			 " A dataset can have many distributions, we will calculate individual score for each distribution and"
+			 +"a distribution gets 5 star if it has non-empty value for property dcat:mediatype or dct:dataformat"
+			 + "We will not check the validity the provided mediatype(IANA type) or dataformat as this is not feasible "
+			 +"finally add all these scores and calculate the averge score"
+			 +"We are doing a ceiling for the score e.g score 2.5 will be returned as 3."
+			 ;
+	
+	
+	@Override
+	public Integer compute(Model model, String datasetUri) throws Exception {
+
+		LOGGER.info("Processing dataset " + datasetUri);
+
+		Resource dataset = ResourceFactory.createResource(datasetUri);
+
+		//Total number of distributions in a dataset.
+				int TotalDistributions = 0;
+				
+				//Store a score for each distribution, we will use it for final evaluation.
+				HashMap<String, Integer> DistributionsAndScores = new HashMap<String, Integer>();	
+				
+				NodeIterator DistributionsIterator = model.listObjectsOfProperty(dataset, DCAT.distribution);
+					
+					while(DistributionsIterator.hasNext()) {
+						
+						/*
+						 * Give 5 stars to a distribution if it has information about 
+						 * MediaType or Dataformat. Else give 0 stars.
+						 */
+						Resource Distribution = (Resource) DistributionsIterator.nextNode();
+						if(Distribution.hasProperty(DCTerms.format))
+						{
+							if(!(Distribution.getProperty(DCTerms.format).getObject().isAnon()) && !(Distribution.getProperty(DCTerms.format).getObject().toString().isEmpty()))
+									DistributionsAndScores.put(Distribution.toString(), 5);
+						}
+						else if(Distribution.hasProperty(DCAT.mediaType)) {
+							
+							if(!(Distribution.getProperty(DCAT.mediaType).getObject().isAnon()) && !(Distribution.getProperty(DCAT.mediaType).getObject().toString().isEmpty()))
+								DistributionsAndScores.put(Distribution.toString(), 5);
+						}
+						else
+							DistributionsAndScores.put(Distribution.toString(), 0);
+						
+						//To calculate how many distributions in a dataset. It will be used for scoring.
+						TotalDistributions++;
+				}
+				
+					/**
+					 * Score Evaluation:
+					 * Total Number of distributions in Dataset = x
+					 * Total aggregated scores of all distributions = y
+					 * Overall Score = y/x
+					 */
+					float AggregatedScoreOfAllDistributions = 0;
+					int OverallScore = 0;
+					for (String key : DistributionsAndScores.keySet()) {
+						AggregatedScoreOfAllDistributions+=DistributionsAndScores.get(key);
+						System.out.println(key +":"+ DistributionsAndScores.get(key));
```

Comment:
remove line. if the metric is applied to 100.000 datasets, 100.000 lines are printed